skrub/_joiner.py

```diff
@@ -1,30 +1,115 @@
 """
-Implements the Joiner, a transformer that allows
-multiple fuzzy joins on a table.
+The Joiner provides fuzzy joining as a scikit-learn transformer.
 """
+import numpy as np
 import pandas as pd
-from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.base import BaseEstimator, TransformerMixin, clone
+from sklearn.compose import make_column_transformer
+from sklearn.feature_extraction.text import HashingVectorizer, TfidfTransformer
+from sklearn.pipeline import make_pipeline
+from sklearn.preprocessing import FunctionTransformer, StandardScaler
 
-from skrub import _join_utils
-from skrub._fuzzy_join import fuzzy_join
+from skrub import _join_utils, _matching
+from skrub._datetime_encoder import DatetimeEncoder
+
+
+def _as_str(column):
+    return column.fillna("").astype(str)
+
+
+DEFAULT_STRING_ENCODER = make_pipeline(
+    FunctionTransformer(_as_str),
+    HashingVectorizer(analyzer="char_wb", ngram_range=(2, 4)),
+    TfidfTransformer(),
+)
+_DATETIME_ENCODER = DatetimeEncoder(resolution=None, add_total_seconds=True)
+
+
+_MATCHERS = {
+    "aux_percentile": _matching.Percentile,
+    "second_neighbor": _matching.QueryNeighbor,
+    "self_join_neighbor": _matching.TargetNeighbor,
+    "worst_match": _matching.MaxDist,
+    "no_rescaling": _matching.Matching,
+}
+DEFAULT_REF_DIST = "aux_percentile"
+
+
+def _make_vectorizer(table, string_encoder, rescale):
+    transformers = [
+        (clone(string_encoder), c)
+        for c in table.select_dtypes(include=["string", "category", "object"]).columns
+    ]
+    num_columns = table.select_dtypes(include="number").columns
+    if not num_columns.empty:
+        transformers.append(
+            (StandardScaler() if rescale else "passthrough", num_columns)
+        )
+    dt_columns = table.select_dtypes("datetime").columns
```

Comment:
I agree but it might be better to have aliases that can be used with the `SelectCols` and `DropCols` (or at least private functions `select` and `drop` rather than global variables representing pandas-specific arguments

Reply:
Yes, for sure